autosklearn/automl.py

```diff
@@ -89,6 +89,7 @@ def __init__(self,
                  get_smac_object_callback=None,
                  smac_scenario_args=None,
                  logging_config=None,
+                 metric=None,
```

Comment:
I don't quite see why you pass the metric in here and do not use it later on.

Reply:
To have it documented here:
We are adding the metric in the class instantiation because the automl object gets created directly during unit testing so that the dummy classifier can be tested.

Not needed during the normal auto-sklearn flow. Long term we would cleanup the fit/related methods because currently both the fit and instantiation takes the metric as argument.